### PR TITLE
Add threads to run arbitrary hoon

### PR DESCRIPTION
**Pull request by ashelkovnykov**
Related to [#197 in urbit/vere](https://github.com/urbit/vere/issues/197).

This PR introduces threads which will be useful to any `conn.c` client. More specifically, these threads will be specifically targeted by the thin and thick clients I plan to produce for the issue linked above.

### CODE REVIEW

**Review by belisarius222 — DISMISSED** [review]
I think in the long run, we'd be better off having this take in a Ford source file that would be parsed to a `$pile` and then built according to the standard Ford build semantics, to allow reuse of Ford runes and recursive imports.  As far as I can tell, there's nothing wrong with this as it is, though.

File: pkg/arvo/sys/hoon.hoon [hunk 1]
```diff
@@ -13231,9 +13231,9 @@
           ;~(pfix jump loan)
         ==
       ==
-   ::
-   ::  parses a or [a b c] or a  b  c  ==
-   ++  lynx
+    ::
```
Comment (belisarius222) [inline pkg/arvo/sys/hoon.hoon:13234]: Let's move this to a separate PR that will go in next time we modify hoon.hoon

**Comment by ashelkovnykov** [comment]
I plan to implement @belisarius222's feedback later in a separate PR.

I originally opened this PR as a draft because I thought I might change the threads slightly while working on the client which uses them. It seems like there's already demand for these threads, so I'll mark this as a regular PR and makes changes in a separate PR.

I'll also move the `hoon.hoon` style fixes to a separate PR so that this one doesn't force a `%base` recompile.

**Review by belisarius222 — APPROVED** [review]
